### PR TITLE
Update sonarcloud.yml file

### PR DESCRIPTION
**Pull request by MaggieKimani1**
- Renames file
- Adds the default branch
- Add dispatch events to allow for manual queue of workflows
- Adds a job for checking the secret and include condition in the build step
- Add path to test coverage files

### CODE REVIEW

**Review by baywet — COMMENTED** [review]
would you mind adding coverlet as well please?

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -43,7 +62,7 @@ jobs:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
         shell: pwsh
         run: |
-          ./.sonar/scanner/dotnet-sonarscanner begin /k:"microsoftgraph_microsoft-graph-devx-api" /o:"microsoftgraph2" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io"
+          ./.sonar/scanner/dotnet-sonarscanner begin /k:"microsoftgraph_microsoft-graph-devx-api" /o:"microsoftgraph2" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.opencover.reportsPaths="tests/**/coverage.opencover.xml"
```
Comment (baywet) [inline .github/workflows/sonarcloud.yml:65]:
```suggestion
          ./.sonar/scanner/dotnet-sonarscanner begin /k:"microsoftgraph_microsoft-graph-devx-api" /o:"microsoftgraph2" /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.opencover.reportsPaths="**/coverage.opencover.xml"
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

File: .github/workflows/sonarcloud.yml [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
+        id: checksecret_job
+        run: |
+            echo "is_SONAR_TOKEN_set: ${{ env.SONAR_TOKEN != '' }}"
+            echo "::set-output name=is_SONAR_TOKEN_set::${{ env.SONAR_TOKEN != '' }}"
   build:
+    needs: [checksecret]
+    if: needs.checksecret.outputs.is_SONAR_TOKEN_set == 'true
     name: Build
     runs-on: ubuntu-latest
     steps:
```
Comment (Ndiritu) [inline .github/workflows/sonarcloud.yml:31]: @MaggieKimani1 Noticed this from the snippet shared on teams. Does DevX-API need to Set up JDK 11?

Comment (Ndiritu) [reply .github/workflows/sonarcloud.yml:31]: My mistake. Realise it's needed by the SonarQube server

**Pull request merged**